src/mono/wasm/debugger/BrowserDebugProxy/DevToolsHelper.cs

```diff
@@ -73,20 +75,22 @@ public static bool TryParse(string id, out DotnetObjectId objectId)
             if (!id.StartsWith("dotnet:"))
                 return false;
 
-            string[] parts = id.Split(":", 3);
+            string[] parts = id.Split(":", 5);
 
             if (parts.Length < 3)
```

Comment:
Please, change to:
```
if (parts.Length < 3 || parts.Length == 4)
```
because when running *DebuggerTests.ArrayTests.InvalidArrayId* we are passing 4 part ID: 
dotnet:array:{ "arrayId": 234980 }. 
After split we are left with 4 elements so the condition is false and we are reaching for non-existing element: parts[4] in the following lines. Without this change the test is failing.

Reply:
I would suggest doing something like https://github.com/dotnet/runtime/blob/release/5.0/src/mono/wasm/runtime/library_mono.js#L508-L512 . 3 parts, `dotnet`, `scheme`, and the `rest`. How to parse the `rest` part would depend on the scheme.